pkg/blockbuilder/scheduler/strategy.go

```diff
@@ -84,6 +90,7 @@ func (p *RecordCountPlanner) Plan(ctx context.Context) ([]*JobWithMetadata, erro
 			jobs = append(jobs, job)
 
 			currentStart = currentEnd
```

Comment:
Not part of this PR but just wanted to check: Does this overlap one record because end of one job == start of next job?

Reply:
yeah, but Max offset of the job is exclusive